### ISSUE TITLE
sqlite unparseable types

### ISSUE THREAD

**Issue by brandonschabell** [sqlfluff/sqlfluff#6434]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When trying to parse a sql statement using the `CHARACTER VARYING` type, I receive an unparseable error.

```sql
CREATE TABLE example (
    id INTEGER PRIMARY KEY,
    description CHARACTER VARYING(32) NOT NULL
);
```

```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE example (\n    id INTEGER PR...'
```

### Expected Behaviour

According to the sqlite documentation, any type with `char` in the name should be associated with TEXT storage type.
Ref: https://www.sqlite.org/datatype3.html#type_affinity

In this case, I'd expect the parsing to be successful, and would further expect no errors to have been detected.

### Observed Behaviour

```sql
CREATE TABLE example (
    id INTEGER PRIMARY KEY,
    description CHARACTER VARYING(32) NOT NULL
);
```

```
$ sqlfluff lint --dialect sqlite .

L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE example (\n    id INTEGER PR...'
```

### How to reproduce

```sql
CREATE TABLE example (
    id INTEGER PRIMARY KEY,
    description CHARACTER VARYING(32) NOT NULL
);
```

```
$ sqlfluff lint --dialect sqlite .

L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE example (\n    id INTEGER PR...'
```

### Dialect

sqlite

### Version

3.2.5

### Configuration

No non-default configuration was used, other than specifying `--dialect sqlite`.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by brandonschabell** [comment]
While the merged PR solves for the specific example I raised, I don't believe it solves the actual issue causing this error- and that is that sqlfluff is being overly prescriptive in terms of which column types are allowed within sqlite. Technically, any type is allowed within sqlite and will be mapped to one of 5 storage classes. Adding an additional allowed column type just masks the issue.
